src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -112,7 +112,9 @@ bool SHAMap::walkBranch (SHAMapTreeNode* node,
     return true;
 }
 
-bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
+bool
+SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
+                 Delta& differences, int maxCount)
```

Comment:
compare isn't a const function? (I tried making it const and got errors, so it is using non-const functions. We might want to look into what's going on here).